packages/catalog-generator/src/main/java/io/kaoto/camelcatalog/generators/CamelCatalogSchemaEnhancer.java

```diff
@@ -0,0 +1,110 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.kaoto.camelcatalog.generators;
+
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.camel.catalog.CamelCatalog;
+import org.apache.camel.tooling.model.EipModel;
+
+import java.util.ArrayList;
+
+public class CamelCatalogSchemaEnhancer {
+
+    private final CamelCatalog camelCatalog;
+
+    public CamelCatalogSchemaEnhancer(CamelCatalog camelCatalog) {
+        this.camelCatalog = camelCatalog;
+    }
+
+    /**
+     * Fill the required properties of the model in the schema if they are not already present
+     *
+     * @param modelName the name of the Camel model
+     * @param modelNode the JSON schema node of the model
+     */
+    void fillRequiredPropertiesIfNeeded(String modelName, ObjectNode modelNode) {
+        EipModel model = camelCatalog.eipModel(modelName);
+        if (model == null) {
+            return;
+        }
+
+        fillRequiredPropertiesIfNeeded(model, modelNode);
+    }
+
+    /**
+     * Fill the required properties of the model in the schema if they are not already present
+     *
+     * @param model     the Camel model
+     * @param modelNode the JSON schema node of the model
+     */
+    void fillRequiredPropertiesIfNeeded(EipModel model, ObjectNode modelNode) {
+        ArrayList<String> requiredProperties = new ArrayList<>();
+
+        if (modelNode.has("required")) {
+            modelNode.get("required").elements().forEachRemaining(node -> {
+                requiredProperties.add(node.asText());
+            });
+        }
+
+        model.getOptions().forEach(option -> {
+            if (option.isRequired() && !requiredProperties.contains(option.getName())) {
+                requiredProperties.add(option.getName());
+            }
+        });
+
+        if (!requiredProperties.isEmpty()) {
+            ArrayNode requiredNode = modelNode.putArray("required");
+            requiredProperties.forEach(requiredNode::add);
+        }
+    }
+
+    /**
+     * Sort schema properties according to the Camel catalog
+     *
+     * @param modelName the name of the Camel model
+     * @param modelNode the JSON schema node of the model
+     */
+    void sortPropertiesAccordingToCatalog(String modelName, ObjectNode modelNode) {
+        EipModel model = camelCatalog.eipModel(modelName);
+        if (model == null) {
+            return;
+        }
+
+        sortPropertiesAccordingToCatalog(model, modelNode);
+    }
+
+    /**
+     * Sort schema properties according to the Camel catalog
+     *
+     * @param model     the Camel model
+     * @param modelNode the JSON schema node of the model
+     */
+    void sortPropertiesAccordingToCatalog(EipModel model, ObjectNode modelNode) {
+        // TODO: Sort properties according to the Camel catalog
```

Comment:
@shivamG640 Here we need to place your Sorting code, and it will be sorting both Schemas and definitions

Reply:
Thanks @lordrip